rules-tests/Php71/Rector/FuncCall/CountOnNullRector/Fixture/skip_in_if_is_countable.php.inc

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Rector\Tests\Php71\Rector\FuncCall\CountOnNullRector\Fixture;
+
+final class SkipInIfIsArray
+{
+    private $data;
+
+    function countables()
+    {
+        $this->data = rand(0, 1)
+            ? []
+            : null;
+
+        if (is_countable($this->data)) {
+            echo count($this->data);
```

Comment:
This looks like simar bug to non empty string.

What is the phpstan type of this variable here?

Reply:
the type is : `PHPStan\Type\Constant\ConstantArrayType` and `PHPStan\Type\ArrayType` by debug:

```php
dump($this->getType($funcCall->args[0]->value));
```

Reply:
In that case, this condition should be ok.

The bug will be in type check then

Reply:
This line should be probably extended with intersection type and similar check.

Reply:
Line 116? the isIterableType() ? 

```php

        if ($this->isAlwaysIterableType($countedType)) {
            return null;
        }
```

Reply:
Yes, now it checks only for union type, but it should include the intersection type too. Maybe the ComplexType node can be used also.

Reply:
The check seems early on `CountableAnalyzer::isCastableArrayType()`, I moved check to it https://github.com/rectorphp/rector-src/pull/1387/commits/521bf6db7d6e6bdeabc8c78ac49c84480772bd50#diff-effb82d9a9df006478a5860e51fd141f60453d9f69b0d89bdef2b5bc0e1b3433